cpu/esp32/esp_xtimer.c

```diff
@@ -93,10 +93,10 @@ void _ets_to_xtimer_free(ETSTimer *timer)
 /* xtimer call back function, distributes ets_timer callbacks */
 void IRAM_ATTR _ets_to_xtimer_callback (void *arg)
 {
-    struct _ets_to_xtimer* e2xt = (struct _ets_to_xtimer*)arg;
+    CHECK_PARAM(e2xt != NULL);
```

Comment:
hmm, is this now checked before that variable is allocated? seems wierd.

What was the CI report?

Reply:
Indeed, I didn't look closely and assumed that it was `CHECK_PARAM(arg != NULL)` as the warning of `cppcheck` makes no sense otherwise.

Here is the output:

```
  	cpu/esp32/esp_xtimer.c:99: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:104: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:121: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:122: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:124: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:125: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:136: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:137: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:147: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:151: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
  	cpu/esp32/esp_xtimer.c:152: warning (nullPointerRedundantCheck): Either the condition 'e2xt!=NULL' is redundant or there is possible null pointer dereference: e2xt.
```

Reply:
I reverted the buggy changes and the cppcheck warnings pop back up. I don't get these locally, so it might be just due to an outdated version of `cppcheck`.